modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/redownload/AlignmentCheck.scala

```diff
@@ -0,0 +1,94 @@
+package org.tessellation.sdk.infrastructure.redownload
+
+import cats.MonadThrow
+import cats.data.NonEmptySet
+import cats.effect.kernel.Temporal
+import cats.effect.std.Supervisor
+import cats.effect.syntax.concurrent._
+import cats.effect.{Async, Ref}
+import cats.syntax.applicative._
+import cats.syntax.eq._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.set._
+
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+import org.tessellation.schema.peer.{L0Peer, Peer, PeerInfo}
+import org.tessellation.schema.{GlobalIncrementalSnapshot, GlobalSnapshotInfo}
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.cluster.storage.{ClusterStorage, L0ClusterStorage}
+import org.tessellation.sdk.domain.seedlist.SeedlistEntry
+import org.tessellation.sdk.domain.snapshot.storage.LastSnapshotStorage
+import org.tessellation.sdk.http.p2p.clients.L0GlobalSnapshotClient
+import org.tessellation.security.hash.Hash
+
+trait AlignmentCheck[F[_]] {
+  def check(): F[Unit]
+}
+
+object AlignmentCheck {
+  private val maxConcurrentPeerInquiries = 10
+  def make[F[_]: Async: Temporal](
+    priorityPeers: NonEmptySet[SeedlistEntry],
+    isAlignedR: Ref[F, Boolean],
+    clusterStorage: ClusterStorage[F],
+    l0ClusterStorage: L0ClusterStorage[F],
+    lastSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+    l0GlobalSnapshot: L0GlobalSnapshotClient[F],
+    waitForPeers: FiniteDuration = 1.minute
+  ): AlignmentCheck[F] = () => {
+//    def findMajority(hashes: NonEmptyList[Hash]): Hash =
+//      hashes.groupMapReduce(identity)(_ => 1).maxBy(_._2)._1
+    def findMajority(hashes: List[Hash]): Option[Hash] =
```

Comment:
There is a `pickMajority` function in `ConsensusStateUpdater` that seems to be doing exactly the same thing. I would suggest extracting it to some object and use in both places.

Reply:
Resolving; code no longer present